### PR TITLE
[AIRFLOW-3615] Parse hostname using netloc

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3615
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

Python 3.5 and 2.7 replace the hostname always with small characters. In the case of a path this is unexpected.

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: tests/models.py [hunk 1]
```diff
@@ -3392,6 +3392,26 @@ def test_connection_from_uri_no_schema(self):
...
 
+    def test_connection_from_uri_without_authinfo(self):
+        uri = 'scheme://host:1234'
+        connection = Connection(uri=uri)
+        self.assertEqual(connection.conn_type, 'scheme')
+        self.assertEqual(connection.host, 'host')
+        self.assertEqual(connection.schema, '')
+        self.assertEqual(connection.login, None)
+        self.assertEqual(connection.password, None)
+        self.assertEqual(connection.port, 1234)
```
Comment (mik-laj) [inline tests/models.py]: This is an additional test not directly related to the issues, but I think it is worth adding it to prevent regression.

**Comment by mik-laj** [comment]
@potiuk  Could you look at this? I launched sytem tests for GCP SQL and everything worked fine. 

@ashb Can I ask you to review?

**Review by potiuk — APPROVED** [review]
Great Change. Fixes important pain-point for some services (CloudSQL for one)

**Review by ashb — APPROVED** [review]
I mean I don't want  to know _why_ you need this, but it looks sensible 😁

File: airflow/models/connection.py [hunk 2]
```diff
@@ -30,6 +30,20 @@
 from airflow.models.base import Base, ID_LEN
 
 
+# Python automatically converts all letters to lowercase in hostname
+# See: https://issues.apache.org/jira/browse/AIRFLOW-3615
+def parse_netlock_to_hostname(uri_parts):
```
Comment (ashb) [inline airflow/models/connection.py]:
```suggestion
def parse_netloc_to_hostname(uri_parts):
```

**Review by ashb — CHANGES_REQUESTED** [review]
Sorry -small typo to fix please.

**Comment by mik-laj** [comment]
@ashb I updated a PR.

**Pull request merged**